src/FlaUI.WebDriver/Controllers/DeviceController.cs

```diff
@@ -0,0 +1,96 @@
+﻿using FlaUI.WebDriver.Models;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.AspNetCore.Mvc.Diagnostics;
+using System.Buffers.Text;
+using System.IO;
+using System.IO.Compression;
+using System.Security.Cryptography;
+using static System.Runtime.InteropServices.JavaScript.JSType;
+
+namespace FlaUI.WebDriver.Controllers
+{
+
+    [Route("session/{sessionId}/appium/device")]
+    [ApiController]
+    public class DeviceController : ControllerBase
+    {
+
+
+        private static ActionResult MissingParameter(string parameterName)
+        {
+            return WebDriverResult.NotFound(new ErrorResponse()
+            {
+                ErrorCode = "Missing JSON Parameter",
+                Message = $"Parameter {parameterName} must be provided in the request."
+            });
+        }
+
+        private static ActionResult FileNotFound(string fileName)
+        {
+            return WebDriverResult.NotFound(new ErrorResponse()
+            {
+                ErrorCode = "File Not Found",
+                Message = $"File {fileName} does not exist."
+            });
+        }
+
+        [HttpPost("push_file")]
+        public async Task<ActionResult> PushFile([FromBody] PushFileRequest request)
+        {
+            if (request.Data == null)
+            {
+                // data could be an empty string, so just a simple null check
+                return MissingParameter("data");
+            }
+            if (string.IsNullOrEmpty(request.Path))
+            {
+                return MissingParameter("path");
+            }
+            var parent = Path.GetDirectoryName(request.Path)!;
+            if (!Directory.Exists(parent))
```

Comment:
To add a little bit of security, we could use a "feature flag" mechanism such as appium-windows-driver to only allow this endpoint if it has been enabled, see https://github.com/appium/appium-windows-driver/blob/master/lib/commands/file-movement.js#L18